src/Microsoft.DotNet.Cli.Utils/CommandResolution/ToolPathCalculator.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Collections.Generic;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class ToolPathCalculator
+    {
+        private readonly string _packagesDirectory;
+
+        public ToolPathCalculator(string packagesDirectory)
+        {
+            _packagesDirectory = packagesDirectory;
+        }
+
+        public string GetBestLockFilePath(string packageId, VersionRange versionRange, NuGetFramework framework)
```

Comment:
Understood. I am actually about to send out a PR with tools in the consuming project's lock file.